modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/actions/model/PreIssueAccessTokenEvent.java

```diff
@@ -54,15 +54,21 @@ public AccessToken getAccessToken() {
 
     @Override
     public boolean equals(Object o) {
-
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
+
         PreIssueAccessTokenEvent that = (PreIssueAccessTokenEvent) o;
-        return Objects.equals(request, that.request) &&
+
+        boolean isEqualGeneral = Objects.equals(request, that.request) &&
                 Objects.equals(accessToken, that.accessToken) &&
-                Objects.equals(tenant, that.tenant) &&
-                Objects.equals(user, that.user) &&
-                Objects.equals(userStore, that.userStore);
+                Objects.equals(tenant, that.tenant);
+
+        if (!"client_credentials".equals(that.request.getGrantType())) {
+            return isEqualGeneral &&
+                    Objects.equals(user, that.user) &&
+                    Objects.equals(userStore, that.userStore);
+        }
```

Comment:
Is this really necessary. If it's client credential user, user store will be null. And this should be verified at validation. Why we want to check for client credentials and behave differently

Reply:
When we are using the `AssertEquals` method of the `TestNG` framework within `testPreIssueAccessTokenActionRequest` test method, it internally invokes the equals method of each nested object. Therefore, during the comparison of `PreIssueAccessTokenActionRequest.PreIssueAccessTokenEvent `object in the `client credentials` grant flow the presence of user-related fields causes the assertion to fail as there is no user object associated. Hence, we have introduced a conditional check to ensure that the unnecessary fields are excluded from the equality determination only for this specific grant type.

As a further improvement, we could consider making User and UserStore Optional fields within the Event class.